### PR TITLE
ui/vccuelist: Slider should change chaser step back to original value…

### PR DESCRIPTION
**Pull request by kwalo**
Hello!

This change fixes bug visualised here:
https://www.youtube.com/watch?v=zBVOZOMIptE&feature=youtu.be

### CODE REVIEW

File: ui/src/virtualconsole/vccuelist.cpp [hunk 1]
```diff
@@ -819,10 +819,12 @@ void VCCueList::slotCurrentStepChanged(int stepNumber)
...
+        int upperBound = 255 - slValue;
+        int lowerBound = qFloor(upperBound - stepVal);
         //qDebug() << "Slider value:" << m_slider1->value() << "Step range:" << (255 - slValue) << (255 - slValue - stepVal);
         // if the Step slider is already in range, then do not set its value
         // this means a user interaction is going on, either with the mouse or external controller
-        if (m_slider1->value() < (255 - slValue - stepVal) || m_slider1->value() > (255 - slValue))
+        if (m_slider1->value() < lowerBound || m_slider1->value() >= upperBound)
         {
             m_slider1->blockSignals(true);
             m_slider1->setValue(255 - slValue);
```
Comment (mcallegari) [inline ui/src/virtualconsole/vccuelist.cpp]: upperBound ?

File: ui/src/virtualconsole/vccuelist.cpp [hunk 2]
```diff
@@ -1149,9 +1151,10 @@ void VCCueList::slotSlider1ValueChanged(int value)
         }
         //qDebug() << "value:" << value << "steps:" << ch->stepsCount() << "new step:" << newStep;
 
-        if (newStep == ch->currentStepIndex())
-            return; // nothing to do
-
+        if (newStep == ch->currentStepIndex()) {
```
Comment (mcallegari) [inline ui/src/virtualconsole/vccuelist.cpp]: Parenthesis on new line, like the rest of the code

**Comment by kwalo** [comment]
I was only testing it with step slider. I'll check it with crossfade
feature and let you know.

Thanks!

sob., 31.03.2018, 10:07 użytkownik Massimo Callegari <
notifications@github.com> napisał:

**Comment by mcallegari** [comment]
In theory the code you changed is in the Step case, so should do no harm to cross fade.
But still, Cue List is a very delicate place where to make changes.
If you address my 2 comments I will merge this.
Thanks

File: ui/src/virtualconsole/vccuelist.cpp [hunk 3]
```diff
@@ -819,10 +819,12 @@ void VCCueList::slotCurrentStepChanged(int stepNumber)
...
 
+        int upperBound = 255 - slValue;
+        int lowerBound = qFloor(upperBound - stepVal);
         //qDebug() << "Slider value:" << m_slider1->value() << "Step range:" << (255 - slValue) << (255 - slValue - stepVal);
         // if the Step slider is already in range, then do not set its value
         // this means a user interaction is going on, either with the mouse or external controller
-        if (m_slider1->value() < (255 - slValue - stepVal) || m_slider1->value() > (255 - slValue))
+        if (m_slider1->value() < lowerBound || m_slider1->value() >= upperBound)
         {
             m_slider1->blockSignals(true);
```
Comment (janosvitok) [inline ui/src/virtualconsole/vccuelist.cpp]: Unrelated comment: This line can be rewritten to `const QSignalBlocker blocker(m_slider1);`; ` m_slider1->blockSignals(false);` down below should be removed.

**Pull request merged**